### PR TITLE
Automated cherry pick of #122193: Keep presence cost to 0 to ensure backward compatibility.

### PR DESCRIPTION
**Pull request by cici37**
Cherry pick of #122193 on release-1.29.

#122193: Keep presence cost to 0 to ensure backward compatibility.

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

```release-note
NONE
```

### CODE REVIEW

**Review by jiahuif — APPROVED** [review]
/lgtm
for backporting compatibility fix.

**Comment by cici37** [comment]
> /lgtm

Hi @jpbetz , would you also tag approve when you have time? This would need your approval before merging

**Comment by jpbetz** [comment]
/approve

This needs a release note and a kind

**Comment by xmudrii** [comment]
Inheriting release note and kind from the original PR
/release-note-edit
```release-note
NONE
```
/kind bug

**Review by xmudrii — APPROVED** [review]
For RelEng:
/lgtm
/approve

**Pull request merged**
